backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -650,21 +650,23 @@ export class ExperimentAssignmentService {
           );
         })
       );
-
-      const allSites = [],
-        allTargets = [];
-      filteredExperiments.forEach((exp) => {
-        exp.partitions.forEach((partition) => {
-          allSites.push(partition.site);
-          allTargets.push(partition.target);
+      let monitoredLogCounts: any[];
+      if (filteredExperiments.some((e) => e.assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS)) {
+        const allWithinSubjectsSites = [],
```

Comment:
small thing, but since I'm already asking for a change, for consistency let's avoid chaining declarations with commas:

```
const allWithinSubjectsSites = [];
const allWithinSubjectsTargets = [];
```